### DISCUSSION TITLE
Random workout

### DISCUSSION THREAD

**Discussion by nareto** [wger-project/wger#998] [Ideas]
I'm a big fan of random workouts: I have several exercise pools, divided by some personal high-level criteria (power, push, pull, core, skill, stretch) and a workout consists of doing one or multiple circuits, every time picking a random exercise from each pool.

Is this something that could be easily implementable? I'm actually a python dev and, time permitting, could have a go at this, though I would like to first coordinate on how to best implement this and **if it is desirable at all**. 

From a fast look at the app I guess the easiest way to do a first implementation of this would be to

1. Make categories editable, so a user can define his own. Alternatively add a new abstraction, like "groups" of exercises
2. Add a "random workout template" abstraction to save the rules to generate a random workout (essentialy just an ordered list of exercises groups)
3. Add functionality to generate `n` workouts  from such a template

**Comment by rolandgeider** [comment]
Hi! Making the workouts more flexible is one of the main features planed for the future (#848) , right now it's not clear how that will look like but one of the possibilities is to implement some kind of templates/scripts that the user could configure and use, such as 5-3-1 with some parameters or preferences. Random workouts would fit well there. The data structure already supports grouping similar exercises like the different squats or curls (but I still need to cleanup the database a bit) which could be used here
